media/server/gstplayer/source/tasks/AttachSource.cpp

```diff
@@ -20,12 +20,126 @@
 #include "tasks/AttachSource.h"
 #include "IGlibWrapper.h"
 #include "IGstWrapper.h"
+#include "IMediaPipeline.h"
 #include "RialtoServerLogging.h"
 
+namespace
+{
+std::string getSampleRateAndChannelsAsCapsStr(const firebolt::rialto::IMediaPipeline::MediaSource &source)
+{
+    std::string capsStr;
+    firebolt::rialto::AudioConfig audioConfig;
+    if (source.getAudioConfig(audioConfig))
+    {
+        if (audioConfig.numberOfChannels != firebolt::rialto::kInvalidAudioChannels)
+            capsStr += ", channels=" + std::to_string(audioConfig.numberOfChannels);
+
+        if (audioConfig.sampleRate != firebolt::rialto::kInvalidAudioSampleRate)
+            capsStr += ", rate=" + std::to_string(audioConfig.sampleRate);
+    }
+
+    return capsStr;
+}
+
+std::string getAudioSpecificConfiguratuionAsCapsStr(std::shared_ptr<firebolt::rialto::server::IGstWrapper> gstWrapper,
+                                                    std::shared_ptr<firebolt::rialto::server::IGlibWrapper> glibWrapper,
+                                                    const firebolt::rialto::IMediaPipeline::MediaSource &source)
+{
+    std::string capsStr;
+    firebolt::rialto::AudioConfig audioConfig;
+    if (source.getAudioConfig(audioConfig) && audioConfig.codecSpecificConfig.size())
+    {
+        gsize codec_priv_size = audioConfig.codecSpecificConfig.size();
+        gconstpointer codec_priv = audioConfig.codecSpecificConfig.data();
+
+        GstCaps *gst_caps = gstWrapper->gstCodecUtilsOpusCreateCapsFromHeader(codec_priv, codec_priv_size);
+        gchar *caps_str = gstWrapper->gstCapsToString(gst_caps);
+        capsStr = caps_str;
+        glibWrapper->gFree(caps_str);
+        gstWrapper->gstCapsUnref(gst_caps);
+    }
+
+    return capsStr;
+}
+
+std::string createCapsStr(std::shared_ptr<firebolt::rialto::server::IGstWrapper> gstWrapper,
+                          std::shared_ptr<firebolt::rialto::server::IGlibWrapper> glibWrapper,
+                          const firebolt::rialto::IMediaPipeline::MediaSource &source)
+{
+    auto mimeType = source.getMimeType();
+    if (mimeType.compare("video/x-h265") == 0)
```

Comment:
I think this is OK for now, but @skyrangelovv @aczs do you think we should put a ticket in the backlog to move the MIME to/from caps mappings to a common place to make maintenance easier?

Reply:
The Firebolt folk are keen on MIME types

Reply:
@skypetts, we definitely should have a ticket for having mime/caps mappers in a common place.

Reply:
Done LLDEV-32732